Library/Formula/ice.rb

```diff
@@ -32,7 +32,10 @@ def install
         --enable-cxx
       ]
 
-      args << "--enable-java" if build.with? "java"
+      if build.with? "java"
+        args << "--enable-java"
+        inreplace "dist/Makefile.in", "@JAVACFLAGS@",  "@JAVACFLAGS@ -source 1.7 -target 1.7"
```

Comment:
TBO I'm not really sure it's an issue with the Berkeley DB build system. The distros which ship it just build with Java 7.

The issue is that if I have Java 7 and 8 installed on my system and I want to build a formula, there's no way to specify a version (unless I missed it in the docs). Setting JAVA_HOME does not work. Brew will always choose Java 8.